src/main/java/org/jahia/modules/appshell/Main.java

```diff
@@ -144,49 +218,119 @@ private void setCustomAttributes(JahiaUser currentUser, HttpServletRequestWrappe
                     resources.add(jsBundle);
                 } else {
-                    resources.add("/modules/" + bundle.getSymbolicName() + "/" + jsBundle);
+                    resources.add("/modules/" + bundleDescriptor.getKey().getSymbolicName() + "/" + jsBundle);
                 }
             }
         }
         return resources;
     }
 
     private List<Bundle> getPackages() {
-        return Arrays.stream(FrameworkService.getBundleContext().getBundles()).filter(bundle -> bundle.getState() == BundleState.ACTIVE.toInt() && BundleUtils.isJahiaModuleBundle(bundle) && (bundle.getResource(PACKAGE_JSON) != null || bundle.getResource(JAHIA_JSON) != null)).collect(Collectors.toList());
+        return Arrays.stream(FrameworkService.getBundleContext().getBundles())
+            .filter(bundle -> bundle.getState() == BundleState.ACTIVE.toInt() &&
+                BundleUtils.isJahiaModuleBundle(bundle) &&
+                (bundle.getResource(PACKAGE_JSON) != null || bundle.getResource(JAHIA_JSON) != null))
+            .collect(Collectors.toList());
     }
 
-    private List<String> getBundleScripts(Bundle bundle, String appName, String key) throws IOException, JSONException {
-        JSONObject pkgJson = new JSONObject(IOUtils.toString(bundle.getResource(PACKAGE_JSON) != null ? bundle.getResource(PACKAGE_JSON) : bundle.getResource(JAHIA_JSON)));
+    private List<String> getBundleScripts(JSONObject pkgJson, String appName, String key) throws IOException, JSONException {
         boolean hasExtend = pkgJson.has(JAHIA) && pkgJson.getJSONObject(JAHIA).has(key) && pkgJson.getJSONObject(JAHIA).getJSONObject(key).has(appName);
         if (hasExtend) {
             Object res = pkgJson.getJSONObject(JAHIA).getJSONObject(key).get(appName);
-            if (res instanceof JSONArray) {
-                JSONArray jsonArray = (JSONArray) res;
-                List<String> l = new ArrayList<>();
-                int s = jsonArray.length();
-                for (int i = 0; i < s; i++) {
-                    l.add(jsonArray.getString(i));
-                }
-                return l;
-            } else {
-                return Collections.singletonList(res.toString());
-            }
+            return getStringList(res);
         }
         return Collections.emptyList();
     }
 
+    private List<String> getStringList(Object jsonObject) throws JSONException {
+        if (jsonObject instanceof JSONArray) {
+            JSONArray jsonArray = (JSONArray) jsonObject;
+            List<String> stringList = new ArrayList<>();
+            for (int i = 0; i < jsonArray.length(); i++) {
+                stringList.add(jsonArray.getString(i));
+            }
+            return stringList;
+        } else {
+            return Collections.singletonList(jsonObject.toString());
+        }
+    }
+
+    private String getSiteFromNode(String nodeUUID, String workspace, Locale locale) {
+        if (nodeUUID == null || nodeUUID.trim().isEmpty()) {
```

Comment:
Could be replace by: `StringUtils.isBlank(nodeUUID)`

Reply:
Good point. I will change it.